src/SquidConfig.h

```diff
@@ -542,6 +550,12 @@ class SquidConfig
         int connect_gap;
         int connect_timeout;
     } happyEyeballs;
+
+    using Id = InstanceId<SquidConfig>;
+    /// A Squid process (re)configuration identifier.
+    /// Changes with each lifecycleStart() call even if
+    /// reconfiguration changed nothing (that Squid could detect).
+    Id id;
```

Comment:
Please consider making this a private `id_` member so that we can protect it from accidental external modifications by returning a _constant_ reference to it.